File: OrbitApi/app/core/password_validation.py
import re


PASSWORD_RULE_MESSAGE = (
    "A senha deve ter entre 8 e 26 caracteres, com letra maiúscula, "
    "letra minúscula, número e caractere especial."
)


def validate_strong_password(value: str) -> str:
    if len(value) < 8 or len(value) > 26:
        raise ValueError(PASSWORD_RULE_MESSAGE)

    checks = [
        re.search(r"[A-Z]", value),
        re.search(r"[a-z]", value),
        re.search(r"\d", value),
        re.search(r"[^A-Za-z0-9]", value),
    ]

    if not all(checks):
        raise ValueError(PASSWORD_RULE_MESSAGE)

    return value

File: Orbit-App/src/utils/passwordValidation.ts
export const passwordRuleMessage =
  "A senha deve ter entre 8 e 26 caracteres, com maiúscula, minúscula, número e caractere especial.";

export function getPasswordValidationError(password: string): string | null {
  if (password.length < 8 || password.length > 26) {
    return passwordRuleMessage;
  }

  const hasUppercase = /[A-Z]/.test(password);
  const hasLowercase = /[a-z]/.test(password);
  const hasNumber = /\d/.test(password);
  const hasSpecial = /[^A-Za-z0-9]/.test(password);

  if (!hasUppercase || !hasLowercase || !hasNumber || !hasSpecial) {
    return passwordRuleMessage;
  }

  return null;
}
